workflow/scripts/infer_window_ts.py

```diff
@@ -0,0 +1,151 @@
+import ray
+import numpy as np
+import tsinfer
+import tskit
+import os
+import zarr
+
+############# this is the code to infer the tree sequence #############
+
+# Get parameters from snakemake
+window_size = snakemake.params.window_size
+output_dir = snakemake.params.output_dir
+window_type = snakemake.params.window_type
+report = snakemake.params.report
+input_vcf = snakemake.input.vcf
+
+# Load ancestral alleles
+ancestral_states = np.load(snakemake.params.ancestral_states)
+# Create output directory
+os.makedirs(output_dir, exist_ok=True)
+
+
+
+# Initialize Ray
+ray.init(logging_level="ERROR")
+
+@ray.remote
+def infer_window_tree_sequence(name, output_dir, window_start, window_end, ds, ancestral_states, site_mask, window_type):
+    """
+    Infer tree sequence for a specific window based on variant indices or physical position
+    
+    Parameters:
+    - name: base name for output files
+    - output_dir: directory to save tree sequences
+    - window_start: start index or position of the window
+    - window_end: end index or position of the window
+    - ds: zarr dataset
+    - ancestral_states: ancestral states array
+    - site_mask: boolean mask for site selection
+    - window_type: type of windowing ('variant_index' or 'physical_position')
+    
+    Returns:
+    - Path to the saved tree sequence file
+    """
+    # Suppress warnings
+    import warnings
+    warnings.simplefilter("ignore")
+    # Create a copy of the site mask
+    current_site_mask = site_mask.copy()
+
+    if window_type == "variant_index":
+        # Mask out variants within the current window based on variant indices
+        window_variant_mask = (np.arange(len(ds.variant_position[:])) >= window_start) & \
+                               (np.arange(len(ds.variant_position[:])) < window_end) 
+        # Temp physical positions for the window based on variant indices
+        temp_physical_position_left = ds.variant_position[window_start]
+        temp_physical_position_right = ds.variant_position[window_end-1] + 1
+        
+    elif window_type == "physical_position":
+        # Mask out variants within the current window based on physical positions
+        window_variant_mask = (ds.variant_position[:] >= window_start) & \
+                               (ds.variant_position[:] < window_end)
+    else:
+        raise ValueError(f"Unknown window_type: {window_type}")
+
+    current_site_mask[window_variant_mask] = False
+
+    # Define the VariantData object
+    window_vdata = tsinfer.VariantData(
+        ds,
+        ancestral_state=ancestral_states[current_site_mask == False],
```

Comment:
See comment above wrt ancestral state -- these are basically just site metadata, not actually used for anything, so seems easier to just use a default character like '0' and avoid the hassle of passing in another file

Reply:
I'm wrong-- these are used to polarize. Passing in ancestral states is the right thing to do, as we're implicitly assuming the data are polarized for the processors, though maybe it'd be more straightforward (on the user side) to pass in a fasta.

Reply:
yep we need this for tsinfer. I believe that the `VariantData` class will such up the `AA` field if it's present in the VCF, but for the test example I'm using `msprime` simulations / VCFs which don't contain it, so we have to pass it this way

Reply:
I think it's better to make this explicitly passed in, anyway, as it's more transparent what's going on. I sort of feel like passing an ancestral fasta is the "easiest" option for the user, but not worth hashing that out now